src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Runtime.Serialization.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.RPC;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Features.BlockStore.Models;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Utilities;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+
+namespace Stratis.Bitcoin.Features.BlockStore.Controllers
+{
+    /// <summary>
+    /// Controller providing operations on a wallet.
+    /// </summary>
+    [Route("api/[controller]")]
+    public class BlockStoreController : Controller
+    {
+        private readonly IBlockStoreCache blockStoreCache;
+
+        /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
+        private readonly Network network;
+
+        private readonly IConnectionManager connectionManager;
+
+        private readonly ConcurrentChain chain;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>Provider of date time functionality.</summary>
+        private readonly IDateTimeProvider dateTimeProvider;
+
+        public BlockStoreController(
+            ILoggerFactory loggerFactory,
+            IBlockStoreCache blockStoreCache,
+            IConnectionManager connectionManager,
+            Network network,
+            ConcurrentChain chain,
+            IBroadcasterManager broadcasterManager,
+            IDateTimeProvider dateTimeProvider)
+        {
+            this.blockStoreCache = blockStoreCache;
+            this.connectionManager = connectionManager;
+            this.network = network;
+            this.chain = chain;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.dateTimeProvider = dateTimeProvider;
+        }
+
+        [Route("block")]
+        [HttpGet]
+        public async Task<IActionResult> GetBlockAsync([FromQuery] ObjectByHashRequest query)
```

Comment:
Maybe call it `GetBlockByHashAsync` if we're going to potentially have a `GetBlockByTransactionAsync`.
Or, instead of having two methods, you could have 1 method `/api/block` that does both.
For that you'd have to change the `ObjectByHashRequest` to be `BlockRequest` (that has both a block hash and a trx hash), and the same controller method can see whether a block hash has been passed or a trx hash has been passed and act appropriately.

Reply:
I will call it GetBlockByHashAsync because I think having 1 single method for 2 things means we need to try one first then if it returns null, try the other one. Potentially every request for transaction hash will happen after a request for block hash (or vice versa), even when you know what you are querying for

Reply:
That' fine but then you'll need 2 endpoints, right?
you can have one method for 2 things if the request has `hash` and `trxHash` for example.
So you would do:
```
if query.Hash
    --find a block by hash
else if query.trxHash
    -- find a block by trx hash
```

Reply:
OK that is good, so for now no changes, and when we want to be able to search by transactionHash, then we'll do it this way

Reply:
Perfect